debezium-server/debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/EventHubsChangeConsumer.java

```diff
@@ -0,0 +1,182 @@
+            throw new DebeziumException(e);
+        }
+
+        LOGGER.info("Using default Event Hubs client for namespace '{}'", producer.getFullyQualifiedNamespace());
+    }
+
+    @PreDestroy
+    void close() {
+        try {
+            producer.close();
+            LOGGER.info("Closed Event Hubs producer client");
+        }
+        catch (Exception e) {
+            LOGGER.warn("Exception while closing Event Hubs producer: {}", e);
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records,
+                            RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        LOGGER.trace("Event Hubs sink adapter processing change events");
+
+        CreateBatchOptions op = new CreateBatchOptions().setPartitionId(partitionID);
+        if (partitionKey != "") {
+            op.setPartitionKey(partitionKey);
+        }
+        if (maxBatchSize.intValue() != 0) {
+            op.setMaximumSizeInBytes(maxBatchSize);
+        }
+
+        EventDataBatch batch = producer.createBatch(op);
+
+        // this loop just adds records to the batch
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            if (null == record.value()) {
+                continue;
+            }
+            EventData eventData = null;
+
+            if (record.value() instanceof String) {
+                eventData = new EventData((String) record.value());
+            }
+            else if (record.value() instanceof byte[]) {
+                eventData = new EventData(getBytes(record.value()));
+            }
+            try {
+                if (!batch.tryAdd(eventData)) {
+                    LOGGER.warn("Event data was too large to fit in the batch - {}", record);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                LOGGER.warn("Event data was null - {}", e.getMessage());
+            }
+            catch (AmqpException e) {
+                LOGGER.warn("Event data is larger than the maximum size of the EventDataBatch - {}", e.getMessage());
+            }
+            catch (Exception e) {
+                LOGGER.warn("Failed to add event data to batch - {}", e.getMessage());
+            }
+        }
+
+        try {
+            producer.send(batch);
+            LOGGER.trace("Sent record batch to Event Hubs");
+        }
+        catch (Exception e) {
+            LOGGER.warn("Failed to send record to Event Hubs {}", e.getMessage());
+        }
+
+        // this loop commits each record
+        for (ChangeEvent<Object, Object> record : records) {
+            try {
+                committer.markProcessed(record);
+                LOGGER.trace("Record marked processed");
+            }
+            catch (Exception e) {
+                LOGGER.warn("Failed to mark record as processed {}", e.getMessage());
```

Comment:
Same here.

Reply:
updated relevant parts to throw `DebeziumException`